### PR TITLE
T31087 lava rest support

### PR DESCRIPTION
**Pull request by mgalka**
- Extract common parts from the LAVA API implementation.
- Create kernelci.lab.lava package
- Move XMLRPC based communication to lab.lava.lava_xmlrpc.
The redesign is a preparation for another implementing other means of  communication with LAVA labs.
- Add lava.lava_rest lab type
- Add hadnling LAVA jobs with LAVA REST API v0.2

### CODE REVIEW

**Comment by gctucker** [comment]
Hitting this error with lab-cip when trying to submit a job:
```
403 Client Error: Forbidden for url: https://lava.ciplatform.org/api/v0.2/jobs/?format=json&limit=256
```
It seems to work fine with lab-collabora though.  Maybe it's a subtle LAVA version difference?

Adding skip label for now to get more jobs to run on staging.

File: config/core/lab-configs.yaml [hunk 1]
```diff
@@ -13,12 +13,12 @@ labs:
...
   lab-broonie:
-    lab_type: lava_rest
+    lab_type: lava.lava_xmlrpc
     url: 'https://lava.sirena.org.uk/'
     priority: low
 
   lab-cip:
-    lab_type: lava_rest
+    lab_type: lava.lava_xmlrpc
     url: 'https://lava.ciplatform.org/'
```
Comment (gctucker) [inline config/core/lab-configs.yaml:22]: Does it need `RPC2` in the URL, for `lava.lava_xmlrpc`?

Comment (mgalka) [reply config/core/lab-configs.yaml]: Yes it does, fixed.

**Review by gctucker — APPROVED** [review]
Tested OK on staging, results are present from all labs:
https://staging.kernelci.org/test/job/kernelci/branch/staging-next/kernel/staging-next-20211112.0/plan/baseline/

**Pull request merged**